gossipsub.go

```diff
@@ -459,6 +499,23 @@ func (gs *GossipSubRouter) handleGraft(p peer.ID, ctl *pb.ControlMessage) []*pb.
 			continue
 		}
 
+		// make sure we are not backing off that peer
+		expire, backoff := gs.backoff[topic][p]
+		if backoff && now.Before(expire) {
+			log.Debugf("GRAFT: ignoring backed off peer %s", p)
+			// refresh the backoff
+			gs.addBackoff(p, topic)
+			// check the flood cutoff -- is the GRAFT coming too fast?
+			floodCutoff := expire.Add(GossipSubGraftFloodThreshold - GossipSubPruneBackoff)
```

Comment:
Another (possibly more correct) way of implementing this could be sending back the desired backoff time to a peer at the time of PRUNE, and penalising peers who do not honour it?

Reply:
Yeah, it would be nice of us to send the backoff period and it makes things more robust for peers with different values of it.
It will take a protocol change however, so let me consider a bit and do it in a follow up PR.

Reply:
follow up in #286.